lib/elixir/src/elixir_bitstring.erl

```diff
@@ -328,6 +339,11 @@ find_match([_Arg | Rest]) ->
 find_match([]) ->
   false.
 
+update_spec_vars(_, _, nomatch) ->
+  nomatch;
+update_spec_vars(#{current_vars := C1}, #{current_vars := C2}, SpecVars) ->
+  maps:merge(SpecVars, maps:filter(fun(K, _V) -> not maps:is_key(K, C1) end, C2)).
```

Comment:
What if instead of computing this as we go, since we may not use it, what if we store the original "E" and instead of checking of SpecVars when find a variable in size, we do something like this instead:

    var in ?key(OriginalE, prematch_vars) or (var in ?key(E, vars) and var notin ?key(OriginalE, vars))

It should achieve the same result and be considerably cheaper. Thoughts?

Reply:
Good idea! 👍

Reply:
Done in https://github.com/elixir-lang/elixir/pull/8598/commits/28f9e17b0c76013a17c2d9b792c19a042c555520.